File: python/candyland/board.py
class Board(object):
	
	def __init__(self):
		self.color_index = 0
		self.build()

	def build(self):
		self.board = []
		self.board_length = 0
		self.color_index = 0
		self.add(4,'','')
		self.add(1,"","Rainbow Trail")
		self.add(3,'','')
		self.add(1,"Pink","Plumpy")
		self.add(8,'','')
		self.add(1,"Pink","Mr Mint")
		self.add(15,'','')
		self.add(1,"","Gumdrop Pass")
		self.add(8,'','')
		self.add(1,"Pink","Jolly")
		self.add(4,'','')
		self.add(1,"","Sticky")
		self.add(26,'','')
		self.add(1,"Pink","Gramma Nutt")
		self.add(10,'','')
		self.add(1,'',"Sticky")
		self.add(9,'','') #95 spaces
		self.add(1,"Pink","Princess Lolly")
		self.add(7,'','')
		self.add(1,'Pink','Queen Frostine')
		self.add(16,'','')
		self.add(1,'','Sticky')
		self.add(12,'','')
		self.add(1,'All','End')


	def add(self,space_count,color,label):
		colors = ["Red","Purple","Yellow","Blue","Orange","Green"]
		start_color = self.color_index  
		for i in range (0,space_count):
			n = (i + start_color -1) % 6
			if ( color and  label):
				space = Space(color,label)
				self.board.append(space)
			else:
				space = Space(colors[n],label)
				self.board.append(space)

			#print "space %s is %s %s" % (self.board_length,space.color,space.label)
			self.color_index = n
			self.board_length = self.board_length + 1


	def get_space_from_position(self,position):
		return self.board[position]


class Space(object):
	
	def __init__(self, color,label):
		#super(Space, self).__init__()
		self.color = color
		self.label = label
	

File: java/candyland/Board.java
package candyland;
import candyland.Space;
import java.util.*;

public class Board {
	int color_index;
	int board_length;
	List<Space> board;

	public Board() {
		this.board  = new ArrayList<Space>();
		this.board_length = 0;
		this.color_index = 0;		
		build();

	}

	public Space get_space_from_positon(int position) {
		return this.board.get(position);

	}

	private void build() {
		

		add_space(4,null,null);
		add_space(1,null,"Rainbow Trail");
		add_space(3,null,null);
		add_space(1,"Pink","Plumpy");
		add_space(8,null,null);
		add_space(1,"Pink","Mr Mint");
		add_space(15,null,null);
		add_space(1,null,"Gumdrop Pass");
		add_space(8,null,null);
		add_space(1,"Pink","Jolly");
		add_space(4,null,null);
		add_space(1,null,"Sticky");
		add_space(26,null,null);
		add_space(1,"Pink","Gramma Nutt");
		add_space(10,null,null);
		add_space(1,null,"Sticky");
		add_space(9,null,null); //95 spaces
		add_space(1,"Pink","Princess Lolly");
		add_space(7,null,null);
		add_space(1,"Pink","Queen Frostine");
		add_space(16,null,null);
		add_space(1,null,"Sticky");
		add_space(12,null,null);
		add_space(1,"All","End");

	}

	private void add_space(int space_count, String color, String label) {
		String[] colors = {"Red","Purple","Yellow","Blue","Orange","Green"};
		int start_color = this.color_index;
		for (int i = 0; i < space_count;i++) {
			int n = (i + start_color) % 6;
			Space space;
			if (color != null && label != null) {
				space = new Space(color,label);
				this.board.add(space);
			}
			else {
				space = new Space(colors[n],label);
				this.board.add(space);
			}
			System.out.println("Space " + this.board_length + " is " + space.color + " " + space.label);
			this.color_index = n;
			this.board_length++;
		}

	}


}